pandas/core/generic.py

```diff
@@ -9711,7 +9713,10 @@ def _where(
             cond = cond.align(self, join="right")[0]
         else:
             if not hasattr(cond, "shape"):
-                cond = np.asanyarray(cond)
+                cond = np.asanyarray(cond, dtype=object)
+            if not cond.flags.writeable:
+                cond.setflags(write=True)
+            cond[isna(cond)] = True
```

Comment:
Doesn't this mutate the caller's `cond` in certain cases? We cannot mutate here.